url86119f019a_solzy_hive_parse_tgz-pJ8-org_apache_hadoop_hive_ql_parse_MainJ8/src/org/apache/hadoop/hive/ql/parse/HiveLexer.java

```diff
@@ -0,0 +1,15495 @@
+			int _channel = DEFAULT_TOKEN_CHANNEL;
+			// HiveLexer.g:308:10: ( '?' )
+			// HiveLexer.g:308:12: '?'
+			{
+			match('?'); 
+			}
+
+			state.type = _type;
+			state.channel = _channel;
+		}
+		finally {
+			// do for sure before leaving
+		}
+	}
+	// $ANTLR end "QUESTION"
+
+	// $ANTLR start "DOLLAR"
+	public final void mDOLLAR() throws RecognitionException {
+		try {
+			int _type = DOLLAR;
+			int _channel = DEFAULT_TOKEN_CHANNEL;
+			// HiveLexer.g:309:8: ( '$' )
+			// HiveLexer.g:309:10: '$'
+			{
+			match('$'); 
+			}
+
+			state.type = _type;
+			state.channel = _channel;
+		}
+		finally {
+			// do for sure before leaving
+		}
+	}
+	// $ANTLR end "DOLLAR"
+
+	// $ANTLR start "JANUARY"
+	public final void mJANUARY() throws RecognitionException {
+		try {
+			int _type = JANUARY;
+			int _channel = DEFAULT_TOKEN_CHANNEL;
+			// HiveLexer.g:315:11: ( 'january' ( 's' )? | 'jan' ( DOT )? )
+			int alt6=2;
+			int LA6_0 = input.LA(1);
+			if ( (LA6_0=='j') ) {
+				int LA6_1 = input.LA(2);
+				if ( (LA6_1=='a') ) {
+					int LA6_2 = input.LA(3);
+					if ( (LA6_2=='n') ) {
+						int LA6_3 = input.LA(4);
+						if ( (LA6_3=='u') ) {
+							alt6=1;
+						}
+
+						else {
+							alt6=2;
+						}
+
+					}
+
+					else {
+						int nvaeMark = input.mark();
+						try {
+							for (int nvaeConsume = 0; nvaeConsume < 3 - 1; nvaeConsume++) {
+								input.consume();
+							}
+							NoViableAltException nvae =
+								new NoViableAltException("", 6, 2, input);
+							throw nvae;
+						} finally {
+							input.rewind(nvaeMark);
+						}
+					}
+
+				}
+
+				else {
+					int nvaeMark = input.mark();
+					try {
+						input.consume();
+						NoViableAltException nvae =
+							new NoViableAltException("", 6, 1, input);
+						throw nvae;
+					} finally {
+						input.rewind(nvaeMark);
+					}
+				}
+
+			}
+
+			else {
+				NoViableAltException nvae =
+					new NoViableAltException("", 6, 0, input);
+				throw nvae;
+			}
+
+			switch (alt6) {
+				case 1 :
+					// HiveLexer.g:315:13: 'january' ( 's' )?
+					{
+					match("january"); 
+
+					// HiveLexer.g:315:25: ( 's' )?
+					int alt4=2;
+					int LA4_0 = input.LA(1);
+					if ( (LA4_0=='s') ) {
+						alt4=1;
+					}
+					switch (alt4) {
+						case 1 :
+							// HiveLexer.g:315:25: 's'
+							{
+							match('s'); 
+							}
+							break;
+
+					}
+
+					}
+					break;
+				case 2 :
+					// HiveLexer.g:315:33: 'jan' ( DOT )?
+					{
+					match("jan"); 
+
+					// HiveLexer.g:315:39: ( DOT )?
+					int alt5=2;
+					int LA5_0 = input.LA(1);
+					if ( (LA5_0=='.') ) {
+						alt5=1;
+					}
+					switch (alt5) {
+						case 1 :
+							// HiveLexer.g:
+							{
+							if ( input.LA(1)=='.' ) {
+								input.consume();
+							}
+							else {
+								MismatchedSetException mse = new MismatchedSetException(null,input);
+								recover(mse);
+								throw mse;
+							}
+							}
+							break;
+
+					}
+
+					}
+					break;
+
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 6277:6332, 6402:6457, 6527:6583, 6653:6708, 6815:6870, 6940:6995, 7065:7120, 7353:7408, 7478:7533, 7603:7658.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.